### ISSUE TITLE
Get note passing editor index 

### ISSUE THREAD

**Issue by Mixonn** [paulrosen/abcjs#282]
Hi Paul!
I am working on a project that will digitalize ethnical songs, but I have a question about abcjs.

1. Is there any list of notes available from Editor (if yes, are they sorted as same as on edit panel? )?
2. Is there any function that for passed char index on the editor will return note id? Example: I want to remove B2, and next note after B2
>(...)  B2 A2c3 CA (..)

So I have to check if A2c4 is whole note, or many notes side by side (I just want to remove next note, not whole "section").
When I will pass char index of *A* (83), the function could return "startChar: 83; endChar:85" for example, so I would know I want to remove the string from 83 to 85. 

AbcJs is really awesome, thank you!

**Comment by paulrosen** [comment]
There isn't a formal api for doing what you want, but that sounds like a good idea! Here's some data you can get; I don't know if this will help:
```
var visualObj = ABCJS.renderAbc(etc...)
console.log(visualObj.lines);
```
That will show you everything available.

You could get individual "events" (that is, either a single rest or note, or a chord, or a bar line) by something like:
```
var ev = visualObj.lines[0].staff[0].voices[0][0]
// returns (for example):
abselem: {info about drawing...}
chord: Array [ {name: "C", position: "default"} ]
duration: 0.25
el_type: "note"
endChar: 48
pitches: Array [ {pitch: 4, verticalPos: 4, highestVert: 10} ]
startChar: 44
```
That is, this is asking for the first staff system, for the first staff on that system, for the first voice on that staff, and for the first event in that voice.

In this example, that was formed from the abc string `"C"G`.

You could iterate through that array and create your own table of startChar and endChar.

Is this what you were looking for?

**Comment by Mixonn** [comment]
Oh, it will really help me, thanks!

So, I am guessing that a similar solution is implemented on the feature, that when I click on Editor (text area), then the corresponding note will become red? (check selection and color every note in selection).

**Comment by paulrosen** [comment]
Here's an example of initializing the editor so that you get a callback:
```
const abcEditor = new abcjs.Editor(
    "textarea-id",
    {
        paper_id: "paper",
        abcjsParams: {
            clickListener: function(abcElem, tuneNumber, classes) { console.log(abcElem, tuneNumber, classes); },
        },
    });
```
`abcElem` will contain the start and end char.

**Comment by Mixonn** [comment]
I am using that Editor class in my project, but I cannot find how does the **"textarea-id"** lets the tune know which note should be "colored to red". 
For example when I click to E2 note in textarea, then the note E2 but in the tune becomes red. 
![image](https://user-images.githubusercontent.com/19841839/70744513-1ebcba80-1d22-11ea-8a03-17477ad212c9.png)
![image](https://user-images.githubusercontent.com/19841839/70744530-267c5f00-1d22-11ea-9195-3cb48b48ae1b.png)

So I guess there is a function that passes cursor index (from textarea) to some rendering service, the **service** is looking for a note on the passed index and then changes the note color to red? Maybe I can get the note this way, like the **service**?

**Comment by paulrosen** [comment]
I see - you are wanting to know when the selection changes in the editor. The clickListener lets you know when the selection changes by clicking the note.

There isn't a built in way, but I'm going to add a callback function for that in the next version.

**Comment by paulrosen** [comment]
Fixed in version 5.11.0
